atlassian_jwt_auth/verifier.py

```diff
@@ -31,15 +40,30 @@ def verify_jwt(self, a_jwt, audience, leeway=0, **requests_kwargs):
         """
         key_identifier = key._get_key_id_from_jwt_header(a_jwt)
         public_key = self._retrieve_pub_key(key_identifier, requests_kwargs)
-
+        public_key_obj = public_key
```

Comment:
Seems like this logic could be in a method that the asyncio verifier could call which would give the benefits of caching for this flow as well.

Reply:
Hrm ... the actual thing that needs to be cached here is the `await public_key` call in the asyncio/verifier.py.  Maybe the verifier should just be caching based on the key_identifier.  But I'm not certain if this is done on purpose or not.

Reply:
> Seems like this logic could be in a method that the asyncio verifier could call which would give the benefits of caching for this flow as well.

Good point, I will adjust this pr.